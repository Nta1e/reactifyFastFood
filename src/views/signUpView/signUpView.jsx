import React, { Component } from "react";
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify'
import SignupForm from "../../components/signUpComponent/signUpComponent";
import signUpAction from '../../actions/signupActions'

export class Signup extends Component {
  state = {
  };
  componentWillReceiveProps(nextProps){
      const { error, success } = nextProps;
      const {history} = this.props;
      if(error.length > 1){
          toast.error(error)
      }else if(success.length > 1){
          toast.success(success)
          setTimeout(()=>{
              history.push('/login')
          }, 3800)
      }

  }
  onChange = (e) =>{
    this.setState({[e.target.name]: e.target.value})
  };
  onSubmit = e =>{
      e.preventDefault();
      const { signUpAction } = this.props;
      const postData = {
          ...this.state
      };
      signUpAction(postData);
  };
  render() {
    return <SignupForm onChange={this.onChange} onSubmit={this.onSubmit} />;
  }
}

Signup.propTypes = {
    error: PropTypes.string.isRequired,
    success: PropTypes.string.isRequired,
    signUpAction: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired
};
export const mapStateToProps = state => ({
   error: state.signup.error,
   success: state.signup.success
});
export default connect(
    mapStateToProps,
    { signUpAction }
)(Signup);
