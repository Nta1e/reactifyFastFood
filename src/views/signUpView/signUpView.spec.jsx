import React from 'react';
import { shallow } from 'enzyme';
import { Signup, mapStateToProps } from './signUpView';

const signUpAction = jest.fn();
const props = {
    error: '',
    success: '',
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    signUpAction
};
const state = {
    signup:{
        error: 'missing fields',
        success: ''
    }
};

describe('test signup view',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<Signup {...props} />)
    });
    it('matches snapshot',()=>{
        expect(wrapper).toMatchSnapshot()
    });
    it('maps state to props', ()=>{
        expect(mapStateToProps(state).error).toEqual('missing fields')
    });
    it('test onchange', ()=>{
        wrapper.instance().onChange({target: {name: 'email', value: 'me@you.com'}});
        expect(wrapper.state('email')).toEqual('me@you.com')
    });
    it('test onSubmit', ()=>{
        wrapper.instance().onSubmit({preventDefault: jest.fn()})
        expect(signUpAction).toBeCalled()

    })

});