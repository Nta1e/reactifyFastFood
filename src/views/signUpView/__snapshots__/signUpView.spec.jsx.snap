// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test signup view matches snapshot 1`] = `ShallowWrapper {}`;
